Sources/MCP/Base/Transports/SSEClientTransport.swift

```diff
@@ -416,7 +416,18 @@ import Logging
 
             // For relative paths, preserve the scheme, host, and port
             let pathToUse = path.starts(with: "/") ? path : "/\(path)"
-            components.path = pathToUse
+            
+            // Parse the path to separate path and query components
+            if let pathComponents = URLComponents(string: pathToUse) {
+                components.path = pathComponents.path
+                // Preserve any query parameters from the endpoint path
+                if let queryItems = pathComponents.queryItems {
+                    components.queryItems = queryItems
+                }
+            } else {
+                components.path = pathToUse
+            }
+            
```

Comment:
Instead of using `URLComponents` a second time here to parse the query, we can split this out with:

```suggestion
            if let queryIndex = pathToUse.firstIndex(of: "?") {
                components.path = String(pathToUse[..<queryIndex])
                let queryString = String(pathToUse[pathToUse.index(after: queryIndex)...])
                components.query = queryString
            } else {
                components.path = pathToUse
            }

```